src/isomorphism.jl

```diff
@@ -0,0 +1,887 @@
+1-element Array{Array{Tuple{Int64,Int64},1},1}:
+ [(3, 1), (1, 2), (2, 3)]
+```
+### See also 
+[`all_induced_subgraphiso`](@ref), [`all_subgraphiso`](@ref), [`has_iso`](@ref), [`count_iso`](@ref)
+"""
+function all_iso(g1::AbstractGraph, g2::AbstractGraph;
+                 vertex_relation::Union{Nothing, Function}=nothing,
+                 edge_relation::Union{Nothing, Function}=nothing,
+                 alg=:vf2)::Channel{Vector{Tuple{eltype(g1),eltype(g2)}}}
+
+    if alg == :vf2
+        make_callback(c) = vmap -> (put!(c, collect(zip(vmap,1:length(vmap)))), return true)
+        T = Vector{Tuple{eltype(g1), eltype(g2)}}
+        ch::Channel{T} = Channel(ctype=T) do c
+            vf2(make_callback(c), g1, g2, IsomorphismProblem, 
+                           vertex_relation=vertex_relation,
+                           edge_relation=edge_relation)
+        end
+        return ch
+    else
+        throw(ArgumentError("Keyword argument alg must be :vf2"))
+    end
+end
+
+"""
+    VF2State{G, T}
+Structure that is internally used by vf2
+"""
+struct VF2State{G, T}
+    g1::G
+    g2::G
+    core_1::Vector{T}
+    core_2::Vector{T}
+    in_1::Vector{T}
+    in_2::Vector{T}
+    out_1::Vector{T}
+    out_2::Vector{T}
+
+    function VF2State(g1::G, g2::G) where {G <: AbstractSimpleGraph{T}} where {T <: Integer}
+        n1 = nv(g1)
+        n2 = nv(g2)
+        core_1 = zeros(T, n1)
+        core_2 = zeros(T, n2)
+        in_1 = zeros(T, n1)
+        in_2 = zeros(T, n2)
+        out_1 = zeros(T, n1)
+        out_2 = zeros(T, n2)
+
+        return new{G, T}(g1, g2, core_1, core_2, in_1, in_2, out_1, out_2)
+    end
+end
+
+"""
+    vf2(callback, g1, g2, problemtype; vertex_relation=nothing, edge_relation=nothing)
+Iterates over all isomorphism between the graphs `g1` (or subgraphs thereof) and `g2`.
+The problem that is solved depends on the value of `problemtype`:
+- IsomorphismProblem: Only isomorphisms between the whole graph `g1` and `g2` are considered.
+- SubGraphIsomorphismProblem: All isomorphism between subgraphs of `g1` and `g2` are considered.
+- InducedSubGraphIsomorphismProblem: All isomorphism between vertex induced subgraphs of `g1` and `g2` are considered.
+
+Upon finding an isomorphism, the function `callback` is called with a vector `vmap` as an argument.
+`vmap` is a vector where `vmap[v] == u` means that vertex `v` in `g2` is mapped to vertex `u` in `g1`.
+If the algorithm should look for another isomorphism, then this function should return `true`.
+
+### Optional Arguments
+- `vertex_relation`: A binary function that takes a vertex from `g1` and one from `g2`. An
+    isomorphism only exists if this function returns `true` for all matched vertices.
+- `edge_relation`: A binary function that takes an edge from `g1` and one from `g2`. An
+    isomorphism only exists if this function returns `true` for all matched edges.
+"""
+function vf2(callback, g1::G, g2::G, problemtype; 
```

Comment:
type of call back can be annotated as Function right?

Reply:
Yeah, I forgot the type annotations there. Thanks.